rooms/dojo.py

```diff
@@ -0,0 +1,345 @@
+from persons.staffs import Staff
+from persons.fellows import Fellow
+from rooms.office import Office
+from rooms.livingspace import LivingSpace
+import random
+import string
+
+
+class Dojo(object):
+    """This is the app main class that has most functions"""
+    def __init__(self):
+        self.all_rooms = {}
+        self.staff_list = []
+        self.fellow_list = []
+        self.allocated = {}
+        self.unallocated = {"office": [], "livingspace" : []}
+
+    def create_room(self, room_name, room_type):
+        """
+        This function create a single room or multiple rooms of the same type
+        by receiving an array of room names an the room type.
+        """
+        log = ""
+        if room_type.strip() != "" and len(room_name) > 0:
+            if room_type.lower() == "office" or \
+                                        room_type.lower() == "livingspace":
+                i = 0
+                for room in room_name:
+                    if room.strip() == "":
+                        log += "\nThe {} at index {} ".format(room_type, \
+                            str(i)) + "cannot be created due to empty name."
+                    elif self.check_room_name_exist(room):
+                        log += "\nThe {} at index {} ".format(room_type, \
+                            str(i)) + "already existed."
+                    else:
+                        self.add_room(room, room_type)
+                    i += 1
+            else:
+                log += "\nCannot create room(s), invalid room type enterred"
+        else:
+            log += "Cannot create rooms with empty " + \
+                                    "room name and/or empty room type"
+        if log == "":
+            return True
+        else:
+            print(log)
+
+    def add_room(self, room_name, room_type):
+        """ This function add the new room to the list of all rooms"""
+        new_room = None
+        if room_type == "office":
+            new_room = Office(room_name)
+            print("An office called {} ".format(room_name) + \
+                                            "has been successfully created")
+        else:
```

Comment:
Is it possible for the `room_type` string to anything else other than `office` or `livingspace`? If so, you could explicitly check if the `room_type` is equivalent to `livingspace`.